server/src/main/java/org/elasticsearch/action/search/SearchPhaseContext.java

```diff
@@ -47,6 +51,22 @@
      */
     SearchRequest getRequest();
 
+    ActionListener<SearchResponse> getListener();
```

Comment:
I exposed these all for convenience to avoid spending too much time refactoring `AbstractSearchAsyncAction`. Instead we should introduce higher-level methods like `cancelSearchTask` or `sendEmptySearchResponse` and avoid exposing these internal details.